src/librustc_typeck/check/wfcheck.rs

```diff
@@ -1277,6 +1293,7 @@ impl ParItemLikeVisitor<'tcx> for CheckTypeWellFormedVisitor<'tcx> {
 
 struct AdtVariant<'tcx> {
     fields: Vec<AdtField<'tcx>>,
+    explicit_discr: Option<DefId>,
```

Comment:
I think a doc-comment here would be nice, even though this is a local struct. Something like:

/// If there is an explicit discriminant on this variant (e.g., `A = 22`), then include the def-id of that expression, so that we can check it is well-formed.